docs/user-guide/src/bmo/reboot_annotation.md

```diff
@@ -0,0 +1,62 @@
+# Reboot annotation
+
+As the title implies, there is an annotation for rebooting a BareMetalHost.
+The annotation key takes either of the following forms:
+
+- `reboot.metal3.io`
+- `reboot.metal3.io/{key}`
+
+In its basic form (`reboot.metal3.io`), the annotation will trigger a reboot of the BareMetalHost.
+The controller will remove the annotation as soon as it has restored power to the host.
+
+The advanced form (`reboot.metal3.io/{key}`) includes a unique suffix (indicated with `{key}`).
+In this form the host will be kept in `PoweredOff` state until the annotation has been removed.
+This can be useful if some tasks needs to be performed while the host is in a known stable state.
+If there are multiple annotations, the controller will wait for all of them to be gone before powering on the host.
+Similarly, if both forms of annotations are used, the `reboot.metal3.io/{key}` form will take precedence.
+
+The annotation value should be a JSON map containing the key `mode` and a value `hard` or `soft` to indicate if a hard or soft reboot should be performed.
+It is not necessary to specify the annotation value.
+In case it is omitted, the default is to first try a soft reboot, and if that fails, do a hard reboot.
+
+Here are a few examples:
+
+- `reboot.metal3.io` - immediate reboot via soft shutdown first, followed by a hard shutdown if the soft shutdown fails.
+- `reboot.metal3.io: {'mode':'hard'}` - immediate reboot via hard shutdown, potentially allowing for high-availability use-cases.
+- `reboot.metal3.io/{key}` - phased reboot, issued and managed by the client registered with the key, via soft shutdown first, followed by a hard reboot if the soft reboot fails.
+- `reboot.metal3.io/{key}: {'mode':'hard'}` - phased reboot, issued and managed by the client registered with the key, via hard shutdown.
```

Comment:
can you add a sentence that shows the purpose of the `key`

Reply:
Good point! Added a bit more about this